exporters/etw/test/etw_tracer_test.cc

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef _WIN32
+
+#  include <gtest/gtest.h>
+#  include <string>
+
+/* TODO: this definition needs to be removed when TraceLoggingDynamic.h is OSS */
+#  ifndef HAVE_NO_TLD
+#    define HAVE_NO_TLD
+#  endif
+
+#  include "opentelemetry/exporters/etw/etw_tracer_exporter.h"
+#  include "opentelemetry/sdk/trace/simple_processor.h"
+
+using namespace OPENTELEMETRY_NAMESPACE;
+
+using ETWEvent = std::map<nostd::string_view, common::AttributeValue>;
+
+TEST(ETWTracer, TracerCheck)
+{
+  std::string providerName = "OpenTelemetry";
+  std::string eventName    = "MyEvent";
+
+  exporter::ETW::TracerProvider tp;
+  auto tracer = tp.GetTracer(providerName);
+  auto span   = tracer->StartSpan("MySpan");
+
+  ETWEvent event = {
+      {"uint32Key", (uint32_t)123456}, {"uint64Key", (uint64_t)123456}, {"strKey", "someValue"}};
+
+  EXPECT_NO_THROW(span->AddEvent(eventName, event));
+  EXPECT_NO_THROW(span->End());
+  EXPECT_NO_THROW(tracer->CloseWithMicroseconds(0));
```

Comment:
It would be great to have an example application illustrating the usage of this exporter, ideally with some instructions on how to receive ETW events. This would help people to use, test and verify the functionality.

Reply:
I will commit the ETW listener (written in C#) to contrib repo. Maybe with example how to use this exporter as forward-channel to C#, that may itself then use OpenTelemetry C# to upload the incoming events.